The unit test TestVolumeBinding is flaky

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Kubernetes CI -->

**Which jobs are flaking**:

The unit test `TestVolumeBinding`

**Which test(s) are flaking**:

```
--- FAIL: TestVolumeBinding (0.71s)
    --- FAIL: TestVolumeBinding/bound_and_unbound_unsatisfied (0.10s)
        volume_binding_test.go:282: call PreFilter and check status
        volume_binding_test.go:291: check state after prefilter phase
        volume_binding_test.go:299: call Filter and check status
        volume_binding_test.go:302: filter status does not match: &{1 [could not find v1.PersistentVolume "pv-a"]}, want: &{3 [node(s) had volume node affinity conflict node(s) didn't find available persistent volumes to bind]}
FAIL
```

**Testgrid link**:

**Reason for failure**:

In volume scheduling, we maintain internal cache which listens on PV/PVC events. However, we don't have a method to wait for them to be synced in UT. 

In production, this is not a problem. Even if the scheduler marks the pod uncacheable due to the stale cache in PVAssumeCache/PVCAssumeCache, the pod will be moved to the active queue later as the scheduler [will move unschedulable pods to the active queue if they are not retried for more than 1 minute](https://github.com/kubernetes/kubernetes/pull/72558).

related:

- https://github.com/kubernetes/kubernetes/pull/73305
- https://github.com/kubernetes/kubernetes/pull/72558

**Anything else we need to know**:
- links to go.k8s.io/triage appreciated
- links to specific failures in spyglass appreciated

--- a/pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go
+++ b/pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go
@@ -252,10 +252,7 @@ func TestVolumeBinding(t *testing.T) {
 				t.Fatal(err)
 			}
 
-			// Start informer factory after initialization
-			informerFactory.Start(ctx.Done())
-
-			// Feed testing data and wait for them to be synced
+			t.Log("Feed testing data and wait for them to be synced")
 			client.StorageV1().StorageClasses().Create(ctx, immediateSC, metav1.CreateOptions{})
 			client.StorageV1().StorageClasses().Create(ctx, waitSC, metav1.CreateOptions{})
 			if item.node != nil {
@@ -267,19 +264,21 @@ func TestVolumeBinding(t *testing.T) {
 			for _, pv := range item.pvs {
 				client.CoreV1().PersistentVolumes().Create(ctx, pv, metav1.CreateOptions{})
 			}
-			caches := informerFactory.WaitForCacheSync(ctx.Done())
-			for _, synced := range caches {
-				if !synced {
-					t.Errorf("error waiting for informer cache sync")
-				}
-			}
 
-			// Verify
+			t.Log("Start informer factory after initialization")
+			informerFactory.Start(ctx.Done())
+
+			t.Log("Wait for all started informers' cache were synced")
+			informerFactory.WaitForCacheSync(ctx.Done())
+
+			t.Log("Verify")
+
 			p := pl.(*VolumeBinding)
 			nodeInfo := framework.NewNodeInfo()
 			nodeInfo.SetNode(item.node)
 			state := framework.NewCycleState()
-			t.Logf("call PreFilter and check status")
+
+			t.Logf("Verify: call PreFilter and check status")
 			gotPreFilterStatus := p.PreFilter(ctx, state, item.pod)
 			if !reflect.DeepEqual(gotPreFilterStatus, item.wantPreFilterStatus) {
 				t.Errorf("filter prefilter status does not match: %v, want: %v", gotPreFilterStatus, item.wantPreFilterStatus)
@@ -288,7 +287,8 @@ func TestVolumeBinding(t *testing.T) {
 				// scheduler framework will skip Filter if PreFilter fails
 				return
 			}
-			t.Logf("check state after prefilter phase")
+
+			t.Logf("Verify: check state after prefilter phase")
 			stateData, err := getStateData(state)
 			if err != nil {
 				t.Fatal(err)
@@ -296,7 +296,8 @@ func TestVolumeBinding(t *testing.T) {
 			if !reflect.DeepEqual(stateData, item.wantStateAfterPreFilter) {
 				t.Errorf("state got after prefilter does not match: %v, want: %v", stateData, item.wantStateAfterPreFilter)
 			}
-			t.Logf("call Filter and check status")
+
+			t.Logf("Verify: call Filter and check status")
 			gotStatus := p.Filter(ctx, state, item.pod, nodeInfo)
 			if !reflect.DeepEqual(gotStatus, item.wantFilterStatus) {
 				t.Errorf("filter status does not match: %v, want: %v", gotStatus, item.wantFilterStatus)

